src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,156 @@
+import { Collection } from './../types/models/Collection';
+import { Item } from './../types/models/Item';
+import { Balance } from '@polkadot/types/interfaces/runtime';
+import { ensureCollection, ensureItem } from './../helpers/verifyUnique';
 import { SubstrateEvent } from "@subql/types";
 import { UniquesTransfer } from "../types/models";
 
 export async function handleUniquesTransferEvent(
   event: SubstrateEvent
-): Promise<void> {
-  logger.debug("uniqueTransferEvent added" + JSON.stringify(event.toHuman()));
+) {
+  logger.debug("uniqueTransferEvent added: " + JSON.stringify(event.toHuman()));
   const from = event.event.data[2];
   const to = event.event.data[3];
   const collectionId = event.event.data[0];
   const itemId = event.event.data[1];
-  const uniqueTransfer = new UniquesTransfer(
-    `${event.block.block.header.number.toNumber()}-${event.idx}`
-  );
-  uniqueTransfer.block = event.block.block.header.number.toNumber();
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
+
+  if (!from || !to || !collectionId || !itemId) {
+    logger.error(
+      "Some arguments is null",
+      JSON.stringify(event.toHuman())
+    );
+    return;
+  }
+
+  const uniqueTransfer = new UniquesTransfer(id);
+
+  uniqueTransfer.block = blockNumber;
   uniqueTransfer.from = from.toString();
   uniqueTransfer.to = to.toString();
-  uniqueTransfer.collectionId = collectionId.toString();
-  uniqueTransfer.itemId = itemId.toString();
   if (event.extrinsic) {
+    event.extrinsic.events.forEach((eventItem) => {
+      if (eventItem.event.method === "Withdraw") {
+        const fee = eventItem.event.data[1];
+        uniqueTransfer.fee = (fee as Balance).toBigInt();
+      }
+    });
     uniqueTransfer.txHash = event.extrinsic.extrinsic.hash.toString();
     uniqueTransfer.timestamp = new Date(
-    event.extrinsic.block.timestamp
-  ).getTime();
+      event.extrinsic.block.timestamp
+    ).getTime();
   }
 
+  const collection = await ensureCollection(collectionId.toString(), id);
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), id);
+  item.owner = to.toString();
+  uniqueTransfer.itemId = item.id;
```

Comment:
`ensureItem` should have set this field, you can remove this line.

Reply:
No, `ensureItem` it's just for validate the item exists if not, then create ir. But can't directly set uniqueTransfer fields